dev/TEAL.md

```diff
@@ -592,23 +614,30 @@ Account fields used in the `acct_params_get` opcode.
 | `b target` | branch unconditionally to TARGET |
 | `return` | use A as success value; end |
 | `pop` | discard A |
+| `popn n` | Remove N values from the top of the stack |
 | `dup` | duplicate A |
 | `dup2` | duplicate A and B |
+| `dupn n` | duplicate A, N times |
 | `dig n` | Nth value from the top of the stack. dig 0 is equivalent to dup |
+| `bury n` | Replace the Nth value from the top of the stack. bury 0 fails. |
 | `cover n` | remove top of stack, and place it deeper in the stack such that N elements are above it. Fails if stack depth <= N. |
 | `uncover n` | remove the value at depth N in the stack and shift above items down so the Nth deep value is on top of the stack. Fails if stack depth <= N. |
+| `frame_dig i` | Nth (signed) value from the frame pointer. |
+| `frame_bury i` | Replace the Nth (signed) value from the frame pointer in the stack |
 | `swap` | swaps A and B on stack |
 | `select` | selects one of two values based on top-of-stack: B if C != 0, else A |
 | `assert` | immediately fail unless A is a non-zero number |
 | `callsub target` | branch unconditionally to TARGET, saving the next instruction on the call stack |
+| `proto a r` | Prepare top call frame for a retsub that will assume A args and R return values. |
 | `retsub` | pop the top instruction from the call stack and branch to it |
+| `switch target ...` | branch to the Ath label. Continue at following instruction if index A exceeds the number of labels. |
 
 ### State Access
 
 | Opcode | Description |
 | - | -- |
-| `balance` | get balance for account A, in microalgos. The balance is observed after the effects of previous transactions in the group, and after the fee for the current transaction is deducted. |
-| `min_balance` | get minimum required balance for account A, in microalgos. Required balance is affected by [ASA](https://developer.algorand.org/docs/features/asa/#assets-overview) and [App](https://developer.algorand.org/docs/features/asc1/stateful/#minimum-balance-requirement-for-a-smart-contract) usage. When creating or opting into an app, the minimum balance grows before the app code runs, therefore the increase is visible there. When deleting or closing out, the minimum balance decreases after the app executes. |
+| `balance` | balance for account A, in microalgos. The balance is observed after the effects of previous transactions in the group, and after the fee for the current transaction is deducted. |
+| `min_balance` | minimum required balance for account A, in microalgos. Required balance is affected by [ASA](https://developer.algorand.org/docs/features/asa/#assets-overview) and [App](https://developer.algorand.org/docs/features/asc1/stateful/#minimum-balance-requirement-for-a-smart-contract) usage. When creating or opting into an app, the minimum balance grows before the app code runs, therefore the increase is visible there. When deleting or closing out, the minimum balance decreases after the app executes. |
```

Comment:
Here we need a link to the specs itself I think, rather than the dev doc, since the later is not versioned the same way.

Also, min-balance is now affected by boxes.

Reply:
I would prefer to remove the references entirely. I would even prefer to remove the entire sentence explaining what causes changes to min_balance.  It doesn't seem like the right place for it.  This opcode accesses a value, we don't have to rehash another part of the spec to explain it. (Just as the the `balance` opcode does not talk about rewards.)

I'd rather focus on the detail that is specific to the opcode: the explanation of timing. That's a question we get in the discord - If I do X, will the AVM see the min balance change immediately?

I'll make a proposed change.